src/Components/Components/src/Rendering/RendererSynchronizationContext.cs

```diff
@@ -6,10 +6,10 @@
 using System.Threading;
 using System.Threading.Tasks;
 
-namespace Microsoft.AspNetCore.Components.Server.Circuits
+namespace Microsoft.AspNetCore.Components.Rendering
 {
     [DebuggerDisplay("{_state,nq}")]
-    internal class CircuitSynchronizationContext : SynchronizationContext
+    internal class RendererSynchronizationContext : SynchronizationContext, IDispatcher
```

Comment:
Now that we're passing the dispatcher into the renderer as a constructor argument, I'm curious about whether it would be possible to keep `RendererSynchronizationContext` inside `Microsoft.AspNetCore.Components.Server`, rather than migrating it to `Microsoft.AspNetCore.Components`.

If we could, that would match the layering pretty cleanly, eliminating it entirely from the WebAssembly case even without the linker.

Reply:
As discussed offline, we want to avoid MVC depending on the server-side bits if possible.